### ISSUE TITLE
Precompiled binary distribution

### ISSUE THREAD

**Issue by monterrr** [DadSchoorse/vkBasalt#16]
It would be great if there where precompiled binaries distributed, particularly a flatpak.

**Comment by DadSchoorse** [comment]
In the release tab are binaries, you just need to unpack them and run make install in the directory.
I do not have the knowledge how to package binaries in any form for any distro.

**Comment by monterrr** [comment]
Could you make a pull request of this software to valve proton to see if they integrate it within proton?

I've made a feature request here: https://github.com/ValveSoftware/Proton/issues/3160

**Comment by monterrr** [comment]
Flatpak's are for GUI apps only, will there be a GUI for this?

**Comment by Calinou** [comment]
If Flatpak sandboxing turns out to be problematic, it's still possible to build a relatively distribution-independent package by using [AppImage](https://appimage.org/) on an old enough base distribution (such as Ubuntu 16.04).

**Comment by WinterSnowfall** [comment]
> I do not have the knowledge how to package binaries in any form for any distro.

While I agree Flatpak/AppImage would probably make more sense in the long run, I've written a packaging script of sorts which could be useful for Debian/Ubuntu folk. The gist of it is that you drop it on top of the current binary package, run it and it churns out a Debian and derivatives compatible .deb package. I'd be happy to share it, if you'd be willing to ship .debs or have any other use for it.

**Comment by Calinou** [comment]
@WinterSnowfall That sounds like [fpm](https://github.com/jordansissel/fpm) or [nfpm](https://github.com/goreleaser/nfpm) to me :wink:

**Comment by WinterSnowfall** [comment]
> @WinterSnowfall That sounds like [fpm](https://github.com/jordansissel/fpm) or [nfpm](https://github.com/goreleaser/nfpm) to me wink

Oh, no, nothing that fancy or general-purpose. I used dpkg-deb and mostly mimicked what `make install` was doing anyway in a bash script, with some changes to deployment dirs (for FHS' sake) and other "quality of life" stuff. I wrote it for personal use (mostly to get vkBasalt running on Ubuntu 18.04, which is currently an adventure), but thought I'd mention it, in case @DadSchoorse is interested.
